src/providers/ldap/sdap_async.c

```diff
@@ -167,6 +195,12 @@ static void sdap_process_result(struct tevent_context *ev, void *pvt)
         return;
     }
 
+    /* Set the chain id if we can match the operation. */
+    op = sdap_get_message_op(sh, msg);
+    if (op != NULL) {
+        sss_chain_id_set(op->chain_id);
```

Comment:
Shouldn't this happen immediately before `sdap_process_message()`?
IIUC, otherwise `tevent_add_timer(sdap_ldap_next_result)` will get `chain_id` of current operations, or do I misunderstand this..?

Reply:
Yes, your suggestion is better. Fixed.